nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -64,8 +64,35 @@ def get_value_for_secret(cls, secret, obj=None, **kwargs):
             msg = f"The secret parameter could not be retrieved for field {err}"
             raise exceptions.SecretParametersError(secret, cls, msg) from err
 
+        # default to token authentication
+        auth_method = "token"
+        if "auth_method" in vault_settings:
+            auth_method = vault_settings["auth_method"]
+
         # Get the client and attempt to retrieve the secret.
-        client = hvac.Client(url=vault_settings["url"], token=vault_settings["token"])
+        if auth_method == "token":
+            try:
+                client = hvac.Client(url=vault_settings["url"], token=vault_settings["token"])
+            except KeyError as err:
+                raise exceptions.SecretProviderError(
+                    secret, cls, "HashiCorp Vault configuration is missing a token"
+                ) from err
+        elif auth_method == "approle":
+            try:
+                client = hvac.Client(url=vault_settings["url"])
+                client.auth.approle.login(
+                    role_id=vault_settings["role_id"],
+                    secret_id=vault_settings["secret_id"],
+                )
```

Comment:
Is there any likely exception we would want to handle in the case where the login credentials are incorrect/rejected by Vault?

Reply:
Good point! Adding now...